common/djangoapps/student/forms.py

```diff
@@ -234,3 +235,51 @@ def cleaned_extended_profile(self):
             for key, value in self.cleaned_data.items()
             if key in self.extended_profile_fields and value is not None
         }
+
+
+class DummyRegistrationExtensionForm(forms.Form):
+    """
+    This form is a placeholder form that can be overwritten by changing the value of
+    settings.REGISTRATION_EXTENSION_FORM. This form should be a standard Django form
+    with a save() method which takes a user argument.
+
+    A normal Django Model form is not quite suitable here, as it likely to require a model
+    which has a foreign key to the user, but the user will not yet be saved, so validation
+    will not complete.
```

Comment:
Is it not possible to use a ModelForm where the user foreign key is `null=True` or `blank=True` ? Wouldn't that be nicer and eliminate a bunch of the complexity here?